pkg/k8s/admission-webhook/validating-webhook_test.go

```diff
@@ -0,0 +1,117 @@
+/*
+    Copyright (C) 2020 Accurics, Inc.
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+package admissionwebhook
+
+import (
+	"io/ioutil"
+	"os"
+	"testing"
+)
+
+func TestAuthorize(t *testing.T) {
+
+	table := []struct {
+		name         string
+		apiKey       string
+		envK8sAPIKey string
+		wantErr      error
+	}{
+		{
+			name:         "empty api key",
+			apiKey:       "",
+			envK8sAPIKey: "valid",
+			wantErr:      ErrAPIKeyMissing,
+		},
+		{
+			name:         "K8S_WEBHOOK_API_KEY env not set",
+			apiKey:       "valid",
+			envK8sAPIKey: "",
+			wantErr:      ErrAPIKeyEnvNotSet,
+		},
+		{
+			name:         "invalid api key",
+			apiKey:       "invalid",
+			envK8sAPIKey: "valid",
+			wantErr:      ErrUnauthorized,
+		},
+		{
+			name:         "valid api key",
+			apiKey:       "valid",
+			envK8sAPIKey: "valid",
+			wantErr:      nil,
+		},
+	}
+
+	for _, test := range table {
+		t.Run(test.name, func(t *testing.T) {
+
+			// validating webhook object
+			var w ValidatingWebhook
+
+			// set K8S_WEBHOOK_API_KEY env if not empty
+			if test.envK8sAPIKey != "" {
+				os.Setenv("K8S_WEBHOOK_API_KEY", test.envK8sAPIKey)
+			}
+			defer os.Unsetenv("K8S_WEBHOOK_API_KEY")
+
+			err := w.Authorize(test.apiKey)
+			if err != test.wantErr {
+				t.Errorf("unexpected error; got: '%v', want: '%v'", err, test.wantErr)
+			}
+		})
+	}
+}
+
+func TestDecodeAdmissionReviewRequest(t *testing.T) {
+
+	table := []struct {
+		name        string
+		requestFile string
+		wantErr     bool
+	}{
+		{
+			name:        "empty review request",
+			requestFile: "testdata/empty.json",
```

Comment:
we should use `filepath.Join()` for all paths

Reply:
Done